pkg/storage/tsdb/caching_bucket.go

```diff
@@ -0,0 +1,73 @@
+package tsdb
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/cache"
+	"github.com/thanos-io/thanos/pkg/cacheutil"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	storecache "github.com/thanos-io/thanos/pkg/store/cache"
+)
+
+type CachingBucketConfig struct {
+	Backend   string                    `yaml:"backend"`
+	Memcached MemcachedIndexCacheConfig `yaml:"memcached"`
```

Comment:
Design wise looks a bit weird reusing `MemcachedIndexCacheConfig`. What do you think if we rename `MemcachedIndexCacheConfig` to `MemcachedClientConfig` (or something generic like that) and we move it away from `pkg/storage/tsdb/index_cache.go`?